### ISSUE TITLE
chore: update GitHub Actions CI for pytest coverage on Python 3.10–3.12

### ISSUE THREAD

**Issue by SilvioBaratto** [SilvioBaratto/stockpy#29]
## Goal
Update `.github/workflows/python-package.yml` to run `pytest --cov=stockpy --cov-report=xml` on Python 3.10, 3.11, and 3.12, upload coverage to Coveralls, and add linting checks (Black, isort, ruff).

## Context
- The current workflow (`python-package.yml`) only tests Python 3.9 and uses `actions/checkout@v2` and `actions/setup-python@v2`.
- `pyproject.toml` already configures Black (88 chars), ruff, pytest, and coverage.
- The project supports Python 3.10+ per `pyproject.toml` `requires-python`.
- Coveralls is already referenced in the existing workflow.

## Agent and skills
- **Primary agent:** `python-pro`
- **Support agents:** none
- **Active skills:** `solid-principles`, `python-expert`
- **Model:** `claude-sonnet-4-6`

## Acceptance criteria
- [ ] Matrix updated to Python 3.10, 3.11, 3.12
- [ ] Uses `actions/checkout@v4` and `actions/setup-python@v5`
- [ ] Installs dependencies from `pyproject.toml` (`pip install -e .[dev]`)
- [ ] Runs `pytest --cov=stockpy --cov-report=xml`
- [ ] Uploads coverage report to Coveralls (with `COVERALLS_REPO_TOKEN` and `GITHUB_TOKEN`)
- [ ] Adds a lint step: `black --check stockpy/` (fails on format violations)
- [ ] Adds a lint step: `isort --check stockpy/` (fails on import order violations)
- [ ] Adds a lint step: `ruff check stockpy/` (fails on lint violations)
- [ ] Workflow triggers on `push` to `main`/`master` and `pull_request`
- [ ] No external secrets needed beyond existing `GITHUB_TOKEN` and `COVERALLS_REPO_TOKEN`

## Dependencies
- None (independent infrastructure task)

## Implementation notes
- The `[dev]` optional dependencies in `pyproject.toml` already include `pytest`, `pytest-cov`, `black`, `ruff`, `coverage`, `pycodestyle`.
- Add `isort` to `[dev]` dependencies if not already present.
- Use `fail-fast: false` in the strategy matrix so all Python versions run even if one fails.
- Continue using `continue-on-error: true` for Coveralls upload if token might be missing.
- Remove the old PEP8 `pycodestyle` step if `ruff` replaces it, or keep it as an additional check.

**Comment by SilvioBaratto** [comment]
Final review (Cycle 3 Phase 5): NOT YET IMPLEMENTED.\n\n still uses:\n- Python 3.9 only (should be 3.10, 3.11, 3.12)\n-  and  (should be v4/v5)\n- ============================= test session starts ==============================
platform linux -- Python 3.12.7, pytest-9.0.3, pluggy-1.6.0
rootdir: /home/silvio/Documents/code/stockpy
configfile: pyproject.toml
testpaths: test
plugins: cov-7.1.0, anyio-4.13.0
collected 223 items

test/test_bigru_forecaster.py ..................                         [  8%]
test/test_bilstm_forecaster.py ...............                           [ 14%]
test/test_callbacks.py ......................                            [ 24%]
test/test_conftest.py ..........                                         [ 29%]
test/test_dataset.py ...........................                         [ 41%]
test/test_dmm_forecaster.py ...................                          [ 49%]
test/test_forecasters.py ....................................            [ 65%]
test/test_gru_forecaster.py .................                            [ 73%]
test/test_lstm_forecaster.py .................                           [ 81%]
test/test_tcn_forecaster.py ...................                          [ 89%]
test/test_transforms.py .......................                          [100%]

=============================== warnings summary ===============================
test/test_callbacks.py::TestLRScheduler::test_step_lr_decreases_learning_rate
test/test_callbacks.py::TestLRScheduler::test_event_name_recorded_in_history
test/test_callbacks.py::TestLRScheduler::test_batch_stepping
  /home/silvio/Documents/code/stockpy/stockpy/callbacks/_lr_scheduler.py:398: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
    lr_scheduler.step()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================= 223 passed, 3 warnings in 6.90s ======================== without \n- PEP8  step (should add Black, isort, ruff)\n- No lint/format checks beyond pycodestyle\n\nRecommendation: Update workflow in a dedicated infra pass.
